### PR TITLE
Refactor and simplify code

### PR DESCRIPTION
**Pull request by intelliot**
Remove dependency on x-address-codec

### CODE REVIEW

File: test/xrp-codec-test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  describe('encodeSeed', function() {
+
+    it('encodes a secp256k1 seed', function() {
+      const result = api.encodeSeed(toBytes('CF2DE378FBDD7E2EE87D486DFB5A7BFF'), 'secp256k1')
+      assert.equal(result, 'sn259rEFXrQrWyx3Q7XneWcwV6dfL')
+    })
+
+    it('encodes low secp256k1 seed', function() {
+      const result = api.encodeSeed(toBytes('00000000000000000000000000000000'), 'secp256k1')
```
Comment (sublimator) [inline test/xrp-codec-test.js]: haha, this test (originally only for ed25519 irrc) was testing out the 3 byte prefix inducement of `sEd`

File: test/codec-test.js [hunk 2]
```diff
@@ -15,11 +15,11 @@ function toBytes(hex) {
 
 describe('ripple-address-codec', function() {
   function makeTest(type, base58, hex) {
-    it('can translate between ' + hex + ' and ' + base58, function() {
+    it('can translate between ' + hex + ' and ' + base58 + ' (encode ' + type + ')', function() {
       const actual = api['encode' + type](toBytes(hex))
       assert.equal(actual, base58)
     })
-    it('can translate between ' + base58 + ' and ' + hex, function() {
+    it('can translate between ' + base58 + ' and ' + hex + ' (decode ' + type + ')', function() {
```
Comment (tpkiddle) [inline test/codec-test.js]: Personally I find template strings more redable and less hassle to write. 😄 
```javascript
`can translate between ${hex} and ${base58} (encode ${type})`
`can translate between ${base58} and ${hex} (decode ${type})`
```

File: src/xrp-codec.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    // for secp256k1, use `FAMILY_SEED`
+    versions: type === 'ed25519' ? ED25519_SEED : [FAMILY_SEED]
+  }
+
+  // prefixes entropy with version bytes
+  return codecWithXrpAlphabet.encode(entropy, opts)
+}
+
+export function decodeSeed(seed: string, opts: {
+  versionTypes?: string[],
```
Comment (tpkiddle) [inline src/xrp-codec.ts]: I'm assuming `versionTypes` doesn't need to be more prescriptive?  i.e. 
`versionTypes?: ['ed25519', 'secp256k1']`

Comment (sublimator) [reply src/xrp-codec.ts]: What's wrong with https://github.com/ripple/x-address-codec ? You don't like to party :) ?

File: tslint.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    "no-empty": true,
+    "no-empty-character-class": true,
+    "no-eval": true,
+    "no-ex-assign": true,
+    "no-extra-boolean-cast": true,
+    "no-extra-semi": true,
+    "no-switch-case-fall-through": true,
+    "no-inner-declarations": [true, "functions"],
+    "no-invalid-regexp": true,
+    // this rule would cause problems with mocha test cases,
```
Comment (jwbusch) [inline tslint.json]: nit: I don't think `.json` formally support inline comments

File: package.json [hunk 5]
```diff
@@ -1,33 +1,33 @@
 {
   "name": "ripple-address-codec",
```
Comment (mDuo13) [inline package.json]: Wish we could rename the repo too, but... oh well.

File: src/xrp-codec.ts [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+/**
+ * XRP codec
+ */
+
+const createHash = require('create-hash')
+
+const NODE_PUBLIC = 28
+// const NODE_PRIVATE = 32
```
Comment (mDuo13) [inline src/xrp-codec.ts]: Why's this commented out?

File: tslint.json (hunk 4 above)
Comment (joeblas) [reply tslint.json]:
```suggestion
    "_comment": "this rule would cause problems with mocha test cases",
```

**Comment by joeblas** [comment]
👍

File: package.json (hunk 5 above)
Comment (intelliot) [reply package.json]: Why can't we? I think we could. Perhaps in a new PR.

**Pull request merged**